src/components/utils/generateStyledComponent.tsx

```diff
@@ -0,0 +1,21 @@
+import React, { ComponentType } from 'react'
+
+import { StyledProps } from '../atoms/types'
+import { generateStyledSystem } from './generateStyledSystem'
+
+import { useTheme } from '~hooks'
+
+type StylePropKeys = 'style' | 'contentContainerStyle'
+
+export const generateStyledComponent =
```

Comment:
Same here, please add some comment with JS doc pattern what is this function doing 🙏